### PR TITLE
fixed subi bug

### PR DESCRIPTION
**Pull request by wanderer**
#82

### CODE REVIEW

File: lib/bn.js [hunk 1]
```diff
@@ -2715,10 +2715,15 @@ BN.prototype.isubn = function isubn(num) {
 
   this.words[0] -= num;
 
-  // Carry
-  for (var i = 0; i < this.length && this.words[i] < 0; i++) {
-    this.words[i] += 0x4000000;
-    this.words[i + 1] -= 1;
+  if (this.words[1] === undefined && this.words[0] < 0) {
```
Comment (indutny) [inline lib/bn.js]: `this.words[1]` is a total garbage (could be a number), if `length` is `1`. I would rather rely on `this.length` instead.

Comment (wanderer) [inline lib/bn.js]: @indutny fixed

File: lib/bn.js [hunk 2]
```diff
@@ -2715,10 +2715,15 @@ BN.prototype.isubn = function isubn(num) {
 
   this.words[0] -= num;
 
-  // Carry
-  for (var i = 0; i < this.length && this.words[i] < 0; i++) {
-    this.words[i] += 0x4000000;
-    this.words[i + 1] -= 1;
+  if (this.words.length === 1 && this.words[0] < 0) {
```
Comment (indutny) [inline lib/bn.js]: `this.words.length` is not representative either, sorry! Please consider using `this.length` instead.

File: test/arithmetic-test.js [hunk 3]
```diff
@@ -136,6 +136,11 @@ describe('BN.js/Arithmetic', function() {
       assert.equal(a.negative, 0);
       assert.equal(a.toString(), '100');
     });
+
+    it('should change sign on small numbers', function() {
+      var a = new BN(0).subn(2)
```
Comment (indutny) [inline test/arithmetic-test.js]: I would suggest adding one more test, like `1 - 2`

**Comment by indutny** [comment]
Looks really great, thank you!

File: test/arithmetic-test.js (hunk 3 above)
Comment (wanderer) [inline test/arithmetic-test.js]: ok done!

File: lib/bn.js (hunk 2 above)
Comment (wanderer) [inline lib/bn.js]: fixed!

Comment (wanderer) [inline lib/bn.js]: fixed!

File: test/arithmetic-test.js [hunk 4]
```diff
@@ -136,6 +136,16 @@ describe('BN.js/Arithmetic', function() {
...
+      assert.equal(a.toString(), '-2');
+    });
+
+    it('should change sign on small numbers', function() {
```
Comment (indutny) [inline test/arithmetic-test.js]: This should have different name.

Comment (wanderer) [inline test/arithmetic-test.js]: done!

**Comment by indutny** [comment]
Thank you, landed and released!

**Pull request closed without merging**